src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java

```diff
@@ -59,13 +63,47 @@
 {
 	@Inject
 	AmahiClient amahiClient;
-
+  ImageView imageView;
+  
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 		setContentView(R.layout.activity_authentication);
 
-		setUpInjections();
+    getUsernameEdit().setOnTouchListener(new View.OnTouchListener() {
+
+            @Override
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                getUsernameEdit().setOnFocusChangeListener(new View.OnFocusChangeListener() {
+                    @Override
+                    public void onFocusChange(View view, boolean b) {
+                        if (b) {
+                            imageView.setImageDrawable(null);
+                        }
+                    }
+                });
+                return false;
+            }
+        });
+
+
+        getPasswordEdit().setOnTouchListener(new View.OnTouchListener() {
+
+            @Override
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                getPasswordEdit().setOnFocusChangeListener(new View.OnFocusChangeListener() {
+                    @Override
+                    public void onFocusChange(View view, boolean b) {
```

Comment:
Same Thing here too. 
Also one more thing you can do is that since the onFocusChangeListeners contain the same code for both the edit texts so you simply make a listener object once and pass it to both the edit texts.